librecval/extract_phrases.py

```diff
@@ -344,20 +345,28 @@ def find_audio_oddities(annotation_path: Path, logger=None) -> Optional[Path]:
     """
 
     # Get folder name without expected track name
-    i = str(annotation_path).rfind("/")
-    _path = str(annotation_path)[:i]
+    _path = str(annotation_path.parent)
+    sound_file = None
 
     # the track number is between the last - and the last .
-    j = str(annotation_path).rfind("-")
-    k = str(annotation_path).rfind(".")
-    track = str(annotation_path)[j + 1 : k]
-    track_1 = track.split("_")[0]
 
-    # find the .wav file
+    match = re.match(
+        r"""
+        .*-(
+            [^-]+   # looking for the string between the last / and the last .
+            )
+            \.[^-.]+
+        """,
+        str(annotation_path),
+        re.VERBOSE,
+    )
+    if match:
+        track = match.group(1)
+        track_1 = track.split("_")[0]
 
-    # try 1: the .wav file is in a subfolder, but it has 'Track number' in it
-    dirs = list(glob.glob(_path + "/**/" + track + "*.wav", recursive=True))
-    sound_file = Path(dirs[0]) if len(dirs) > 0 and Path(dirs[0]).exists() else None
+        # try 1: the .wav file is in a subfolder, but it has 'Track number' in it
+        dirs = list(glob.glob(_path + "/**/" + track + "*.wav", recursive=True))
```

Comment:
```suggestion
        dirs = list(_path.glob("/**/" + track + "*.wav"))
```

Reply:
PATHS FOR THE PATHLIB CROWN! F-STRINGS FOR THE F-STRING HORDE

```suggestion
        dirs = list(_path.glob(f"**/{track}/*.wav"))
```

https://docs.python.org/3/library/pathlib.html#pathlib.Path.glob

Reply:
Do these changes keep the "recursive" property? I remember it being a pain to figure out that I needed that argument

Reply:
Yup, the link says,

> The “**” pattern means “this directory and all subdirectories, recursively”. In other words, it enables recursive globbing

Reply:
Oh perfect, thank you!
I was using `glob.glob` in a lot of places...